services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,19 +43,44 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  xQueueSend(thermalMgrQueueHandle, event, 0);
   return ERR_CODE_SUCCESS;
+  
 }
 
 void osHandlerLM75BD(void) {
+  
   /* Implement this function */
+  
+  float *temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, temp);
+  if (*temp > 75.0) {
+    overTemperatureDetected();
+  }
+  else {
+    safeOperatingConditions();
+  } 
```

Comment:
`readTempLM75BD` is a blocking function call. Internally, it'll try to take a mutex that protects the I2C bus. So you shouldn't call it from `osHandlerLM75BD` since this function is meant to execute from within an ISR.